src/client/js/data/fetching.test.js

```diff
@@ -0,0 +1,52 @@
+const {fetchData} = require('./fetching')
+
+describe('fetchData', () => {
+  let mockedSuccess = true
+  let mockedPayload = {}
+  let mockedResponse = {
+    ok: mockedSuccess,
+    ...mockedPayload,
+    json: function() {
+      return mockedPayload
+    }
+  }
+
+  beforeEach(() => {
+    global.fetch = jest.fn().mockImplementation(() => {
+      return new Promise((resolve, reject) => {
+        resolve(mockedResponse)
+      })
+    })
+  })
+
+  it('returns response json', async () => {
+    mockedSuccess = true
+    mockedPayload = { foo: 'bar' }
+
+    await expect(fetchData()).resolves.toEqual(mockedPayload)
+  })
+
+  it('throws if response error', async () => {
+    mockedSuccess = false
+
+    try {
+      await expect(fetchData())
+    } catch (e) {
+      expect(e.code).toEqual('HTTP_ERROR')
+    }
+  })
+
+  it('calls api with appropriate url', async () => {
+    mockedSuccess = true
+    mockedPayload = { foo: 'bar' }
+
+    await fetchData()
+    expect(global.fetch).toHaveBeenCalledWith('/api/stories?filter=undefined&')
```

Comment:
Does not look correct but left to confirm

Reply:
It looks like `fetchData` expects a filter to be passed in as the first argument, perhaps there needs to be more validation in the function, what do you think?

Reply:
I'll rewrite for clarity, just wanted to confirm it was not intended